plugins/http2_adapter/lib/src/connection_manager_imp.dart

```diff
@@ -45,12 +45,13 @@ class _ConnectionManager implements ConnectionManager {
       if (initFuture == null) {
         _connectFutures[domain] = initFuture = _connect(options);
       }
-      transportState = await initFuture;
+      transportState = await initFuture.whenComplete(() {
+         _connectFutures.remove(domain);
```

Comment:
Can we remove this only when the future failed?

Reply:
I changed code only removed cached Future on fail and restored the logic of success to its original state.